test/middleware-config/express-entry.ts

```diff
@@ -33,22 +34,19 @@ async function startServer() {
     app.use(viteDevMiddleware)
   }
 
-  const middlewares = (await getMiddlewares()) as Middleware[]
-  middlewares.sort((m1, m2) => getOrder(m1.order) - getOrder(m2.order))
-  middlewares.forEach((middlewareSpec) => {
-    const middleware = middlewareSpec.value
-    app.all('*', createMiddleware(() => middleware)())
-  })
+  const middlewares = (await getMiddlewares()) as (RouteDefinition | MiddlewareDefinition)[]
+  const router = new UniversalRouter()
+  apply(router, middlewares)
+  app.all('*', createMiddleware(() => router[universalSymbol])())
+  // TODO replace with UniversalExpressRouter once done.
+  //      It should look like this:
+  // const router = new UniversalExpressRouter(app)
+  // apply(router, middlewares)
+  //      So no need for manually calling app.*, and rely on express routing
```

Comment:
This could be moved to `vike-node`, which would be responsible to create the proper `Universal*Router`.

Reply:
:+1: And eventually `vike-{express,hono,...}` so that one Vike extension = one tool (e.g. to fit the copy next to the slot machine on the landing page).

Reply:
It should be simple enough, and I like the idea.
I can see that in Bati we would then just add a comment somewhere or a paragraph in the README to explain that it can be ejected.
We could even allow the user to retrieve the built-in express like this:
```ts
import { app } from "vike-express";
```

Reply:
:+1:

> We could even allow the user to retrieve the built-in express like this:

Neat idea :sparkles: